pkg/services/ngalert/notifier/channels/slack.go

```diff
@@ -339,3 +400,59 @@ func (sn *SlackNotifier) buildSlackMessage(ctx context.Context, as []*types.Aler
 func (sn *SlackNotifier) SendResolved() bool {
 	return !sn.GetDisableResolveMessage()
 }
+
+func (sn *SlackNotifier) slackFileUpload(ctx context.Context, data io.Reader, recipient, token string) error {
+	sn.log.Info("Uploading to slack via file.upload API")
+	headers, uploadBody, err := sn.generateSlackBody(data, token, recipient)
+	if err != nil {
+		return err
+	}
+	cmd := &models.SendWebhookSync{
+		Url: "https://slack.com/api/files.upload", Body: uploadBody.String(), HttpHeader: headers, HttpMethod: "POST",
+	}
+	if err := sn.webhookSender.SendWebhookSync(ctx, cmd); err != nil {
+		sn.log.Error("Failed to upload slack image", "error", err, "webhook", "file.upload")
+		return err
+	}
+	return nil
+}
+
+func (sn *SlackNotifier) generateSlackBody(data io.Reader, token string, recipient string) (map[string]string, bytes.Buffer, error) {
```

Comment:
```suggestion
func (sn *SlackNotifier) generateFileUploadBody(data io.Reader, token string, recipient string) (map[string]string, bytes.Buffer, error) {
```
Or something similar. Makes it obvious this isn't the body for the main notification but rather for file uploads. We can drop the `Slack` segment because we're in the `slack.go` file to make it shorter.